nemo/core/classes/modelPT.py

```diff
@@ -444,6 +445,7 @@ def restore_from(
         override_config_path: Optional[str] = None,
         map_location: Optional[torch.device] = None,
         strict: bool = False,
+        quantize: bool = False,
```

Comment:
We can avoid such a deep integration with the core by refactoring out some component of restore_from - mainly its config loading and config overriding. We can add a method to extract and instantiate just the model's config (and not the model itself) and return that - this will give you a model's config prior to instantiation to inject the `model.*.quantize = True` flag and then restore from using this overriden config. 

The primary reason for doing this is 
 - It is generic and allows overriding parts of the model before restoring weights (as long as changes dont affect weights themselves)
 - quantize is a property of just 1 model right now and not general to all Nemo models, so it doesnt seem right to expose the `quantize=False` flag in core methods across all models.
 - The way quantize is being propagated right now, it will prevent instantiation of any other model since no other NeMo model constructor has `quantize` argument in it